packages/explorer-helpers/src/constructMultipartForm.ts

```diff
@@ -0,0 +1,102 @@
+import type { JSONValue } from './types';
+
+export type FileVariable = {
+  variableKey: string;
+  files: { arrayBuffer: ArrayBuffer; fileName: string }[];
+  isMultiFile: boolean;
+};
+
+// https://github.com/apollographql/apollo-client/blob/cbcf951256b22553bdb065dfa0d32c0a4ca804d3/src/link/http/serializeFetchParameter.ts
+export const serializeFetchParameter = (p: any, label: string) => {
+  let serialized;
+  try {
+    serialized = JSON.stringify(p);
+  } catch (e) {
+    const parseError = new Error(
+      `Network request failed. ${label} is not serializable`
+    );
+    throw parseError;
+  }
+  return serialized;
+};
+
+export const constructMultipartForm = async ({
+  fileVariables: inputtedFileVariables,
+  requestBody,
+}: {
+  fileVariables: FileVariable[];
+  requestBody: {
+    operationName?: string;
+    query: string;
+    variables?: Record<string, string>;
+  };
+}) => {
+  const fileVariables: {
+    variableKey: string;
+    files: File[];
+    isMultiFile: boolean;
+  }[] = inputtedFileVariables.map((fileVariable) => ({
+    ...fileVariable,
+    files: fileVariable.files.map(
+      ({ arrayBuffer, fileName }) =>
+        new File([new Blob([arrayBuffer])], fileName)
+    ),
```

Comment:
Ahh this is a bit unique here, in studio we expect these to already be files. So maybe this should be moved to a helper, and  the mapping should happen before passing the variables to `constructMultipartForm`?